cmake/defaults/gccclangshareddefaults.cmake

```diff
@@ -42,8 +42,14 @@ endif()
 _disable_warning("deprecated")
 _disable_warning("deprecated-declarations")
 
-# Suppress unused typedef warnings emanating from boost.
-_disable_warning("unused-local-typedefs")
+# Suppress unused typedef warnings eminating from boost.
```

Comment:
So what I'm really trying to get at is I think the check ought to be if (!clang || clang >= 3.6), since no-unused-local-typedefs was introduced in 3.6. The check for (clang > 3.5) allows 3.5.x to pass, which I don't think is correct.

Unfortunately, cmake doesn't provide a VERSION_GREATER_OR_EQUAL, so "NOT ... VERSION_LESS 3.6" was the most succinct alternative.